### PR TITLE
Add per-qubit duration in a scheduled circuit

### LINKED ISSUE #5109: per-qubit duration in a scheduled circuit
ability to inquire the start_time and stop_time of **each qubit** in a circuit that has been scheduled. This is doable in pulse schedules (per channel) and is very useful when trying to find critical paths.

### PR DESCRIPTION
**Pull request by itoko**
### Summary
Fix #5109 

### Details and comments
Add functions to inquire the `start_time` and `stop_time` of each qubit in a circuit that has been scheduled.

- `qubit_start_time`: the start time of the first instruction, excluding delays, over the supplied qubits.
- `qubit_stop_time`: the stop time of the last instruction, excluding delays, over the supplied qubits.
- `qubit_duration`: `qubit_stop_time - qubit_start_time`, which is the same spec as the future spec of `pulse.Schedule.ch_duration` after #4848.

### CODE REVIEW

File: qiskit/circuit/quantumcircuit.py [hunk 1]
```diff
@@ -2369,6 +2370,93 @@ def add_calibration(self, gate, qubits, schedule, params=None):
...
+
+        Returns:
+            Return the start time of the first instruction, excluding delays, over the qubits
+
+        Raises:
+            CircuitError: if ``self`` is a not-yet scheduled circuit.
+        """
+        from .delay import Delay
+        if self.duration is None:
+            raise CircuitError("bit_start_time is defined only for scheduled circuit.")
```
Comment (ajavadia) [inline qiskit/circuit/quantumcircuit.py:2404]:
```suggestion
            raise CircuitError("qubit_start_time is defined only for scheduled circuit.")
```

File: qiskit/circuit/quantumcircuit.py [hunk 2]
```diff
@@ -2369,6 +2370,93 @@ def add_calibration(self, gate, qubits, schedule, params=None):
...
+            *qubits: Qubits within ``self`` to include. Integers are allowed for qubits, indicating
+            indices of ``self.qubits``.
+
+        Returns:
+            Return the start time of the first instruction, excluding delays, over the qubits
+
+        Raises:
+            CircuitError: if ``self`` is a not-yet scheduled circuit.
+        """
+        from .delay import Delay
```
Comment (ajavadia) [inline qiskit/circuit/quantumcircuit.py:2402]: can't you import this at the top? (similar for the import in `qubit_stop_time`

File: qiskit/circuit/quantumcircuit.py [hunk 3]
```diff
@@ -2369,6 +2370,93 @@ def add_calibration(self, gate, qubits, schedule, params=None):
...
+
+        Returns:
+            Return the stop time of the last instruction, excluding delays, over the qubits
+
+        Raises:
+            CircuitError: if ``self`` is a not-yet scheduled circuit.
+        """
+        from .delay import Delay
+        if self.duration is None:
+            raise CircuitError("bit_start_time is defined only for scheduled circuit.")
```
Comment (ajavadia) [inline qiskit/circuit/quantumcircuit.py:2441]:
```suggestion
            raise CircuitError("qubit_start_time is defined only for scheduled circuit.")
```

**Review by ajavadia — COMMENTED** [review]
looks good. why do you allow multiple qubits as args to `qubit_duration`, `qubit_start_time` and `qubit_stop_time`? the duration of each qubit seems to be independent to me.

**Comment by itoko** [comment]
No strong reason, I just followed the conventions of `pulse.Schedule.ch_duration` family, which allow multiple channels. Although I have no strong preference, they would help to save a few typings:
```
max(qubit_stop_time(q) for q in qubits) - min(qubit_start_time(q) for q in qubits)
=>
qubit_duration(*qubits)
```

**Review by ajavadia — APPROVED** [review]

**Comment by ajavadia** [comment]
ok let's keep it

**Pull request merged**